### PR TITLE
Reduce mutexes in Auth module

### PR DESCRIPTION
**Pull request by Pietfried**
## Describe your changes
Reduce mutexes in Auth module:
* Only using event_mutex for public functions that can be called from different threads
* Unlocking before select_evse and lock only referenced_evse mutexes in it

This change leads to less complexity and reduced the number of mutexes

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: modules/Auth/lib/ReservationHandler.cpp [hunk 1]
```diff
@@ -253,19 +248,14 @@ ReservationHandler::cancel_reservation(const int reservation_id, const bool exec
 
     std::pair<bool, std::optional<uint32_t>> result;
 
-    std::unique_lock<std::recursive_mutex> lock(reservation_mutex);
```
Comment (maaikez) [inline modules/Auth/lib/ReservationHandler.cpp:256]: This function is also called from the timer within the reservation handler (see line 581, in 'set_reservation_timer'). So I think the mutex should not be removed here.

Comment (Pietfried) [reply modules/Auth/lib/ReservationHandler.cpp:256]: I added that one again

**Review by maaikez — APPROVED** [review]

File: modules/Auth/tests/auth_tests.cpp [hunk 2]
```diff
@@ -198,6 +199,147 @@ TEST_F(AuthTest, test_two_referenced_connectors) {
...
+
+    ASSERT_TRUE(result1 == TokenHandlingResult::TIMEOUT);
+    ASSERT_TRUE(result2 == TokenHandlingResult::TIMEOUT);
+    ASSERT_FALSE(this->auth_receiver->get_authorization(0));
+    ASSERT_FALSE(this->auth_receiver->get_authorization(1));
+    ASSERT_FALSE(this->auth_receiver->get_authorization(2));
+    ASSERT_FALSE(this->auth_receiver->get_authorization(3));
+    ASSERT_FALSE(this->auth_receiver->get_authorization(4));
+
+    EVLOG_critical << "HQH";
```
Comment (hikinggrass) [inline modules/Auth/tests/auth_tests.cpp:340]: hqh? :smile:

**Review by hikinggrass — APPROVED** [review]

File: modules/Auth/tests/auth_tests.cpp (hunk 2 above)
Comment (Pietfried) [inline modules/Auth/tests/auth_tests.cpp:340]:
```suggestion
```

**Pull request merged**